applications/serial_lte_modem/src/lwm2m_carrier/slm_at_carrier.c

```diff
@@ -861,6 +953,647 @@ int handle_at_carrier(enum at_cmd_type cmd_type)
+
+	int ret;
+	uint16_t enabled;
+
+	ret = at_params_unsigned_short_get(&slm_at_param_list, 2, &enabled);
+	if (ret) {
+		return ret;
+	}
+
+	if (enabled > 1) {
+		LOG_DBG("AT#XCARRIERCFG=\"bootstrap_smartcard\" failed: must be 0 or 1");
+		return -EINVAL;
+	}
+
+	return lwm2m_settings_bootstrap_from_smartcard_set(enabled);
+}
+
+const static uint32_t carriers_enabled_map[] = {
+	LWM2M_CARRIER_GENERIC,
+	LWM2M_CARRIER_VERIZON,
+	LWM2M_CARRIER_ATT,
+	LWM2M_CARRIER_LG_UPLUS,
+	LWM2M_CARRIER_T_MOBILE,
+	LWM2M_CARRIER_SOFTBANK
+};
+
+static const char *carriers_enabled_str(void)
+{
+	static char oper_str[12] = { 0 };
+	uint32_t carriers_enabled = lwm2m_settings_carriers_enabled_get();
+	int offset = 0;
+
+	if (carriers_enabled == UINT32_MAX) {
+		return "all";
+	} else if (carriers_enabled == 0) {
+		return "";
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(carriers_enabled_map); i++) {
+		if ((carriers_enabled & carriers_enabled_map[i]) && (offset < sizeof(oper_str))) {
+			offset += snprintf(&oper_str[offset], sizeof(oper_str) - offset, "%s%u",
+					   (offset == 0) ? "" : ", ", i);
+		}
+	}
+
+	return oper_str;
+}
+
+/* AT#XCARRIERCFG="carriers"[,"all"|<carrier1>[<carrier2>[,...[,<carrier6>]]]] */
+static int do_cfg_carriers(void)
+{
+	uint32_t count;
+
+	count = at_params_valid_count_get(&slm_at_param_list);
+	if (count == 2) {
+		rsp_send("\r\n#XCARRIERCFG: %s\r\n", carriers_enabled_str());
+		return 0;
+	}
+
+	uint32_t carriers_enabled = 0;
+	int ret, size = sizeof("all");
+	char buf[size];
+
+	if (count - 2 > ARRAY_SIZE(carriers_enabled_map)) {
+		LOG_DBG("AT#XCARRIERCFG=\"carriers\" failed: too many parameters");
+		return -EINVAL;
+	}
+
+	ret = util_string_get(&slm_at_param_list, 2, buf, &size);
+	if (!ret && slm_util_casecmp(buf, "all")) {
+		carriers_enabled = UINT32_MAX;
+	} else {
+		for (int i = 2; i < count; i++) {
+			uint16_t carrier;
+
+			ret = at_params_unsigned_short_get(&slm_at_param_list, i, &carrier);
+			if (ret || (carrier >= ARRAY_SIZE(carriers_enabled_map))) {
+				LOG_DBG("AT#XCARRIERCFG=\"carriers\" failed: illegal operator");
+				return -EINVAL;
+			}
+
+			carriers_enabled |= carriers_enabled_map[carrier];
+		}
+	}
+
+	return lwm2m_settings_carriers_enabled_set(carriers_enabled);
+}
+
+/* AT#XCARRIERCFG="coap_con_interval"[,<interval>] */
+static int do_cfg_coap_con_interval(void)
+{
+	uint32_t count;
+
+	count = at_params_valid_count_get(&slm_at_param_list);
+	if (count == 2) {
+		rsp_send("\r\n#XCARRIERCFG: %d\r\n", lwm2m_settings_coap_con_interval_get());
+		return 0;
+	}
+
+	int ret, coap_con_interval;
+
+	ret = at_params_int_get(&slm_at_param_list, 2, &coap_con_interval);
+	if (ret) {
+		return ret;
+	}
+
+	if (coap_con_interval < -1 || coap_con_interval > 86400) {
+		LOG_DBG("AT#XCARRIERCFG=\"coap_con_interval\" failed: value must be "
+			"-1 or between 0 and 86400 (24 hours)");
+		return -EINVAL;
+	}
+
+	return lwm2m_settings_coap_con_interval_set(coap_con_interval);
+}
+
+/* AT#XCARRIERCFG="config_enable"[,<0|1>] */
```

Comment:
Should this and others like this start with custom_? Similarly to how set and get are named in lwm2m.